src/event_pusher/mod_event_pusher_push.erl

```diff
@@ -108,10 +112,14 @@ stop(Host) ->
 %%--------------------------------------------------------------------
 -spec push_event(Acc :: mongoose_acc:t(), Host :: jid:lserver(),
                  Event :: mod_event_pusher:event()) -> mongoose_acc:t().
-push_event(Acc, Host, Event = #chat_event{type = Type, direction = out, to = To}) when Type =:= groupchat;
-                                                                                       Type =:= chat ->
+push_event(Acc, Host, Event = #chat_event{direction = out, to = To,
+                                          type = Type}) when Type =:= groupchat;
+                                                             Type =:= chat ->
     BareRecipient = jid:to_bare(To),
     do_push_event(Acc, Host, Event, BareRecipient);
+push_event(Acc, Host, Event = #unack_msg_event{user = U, server = S}) ->
+    BareRecipient = jid:make(U, S, <<"">>),
+    do_push_event(Acc, Host, Event, BareRecipient);
```

Comment:
Precisely, as I said in the comment in `enhanced`, here's a case of stringprepping things twice. `do_push_event` will eventually arrive to `should_publish`, which will do `jid:make` again.

Reply:
+1 from me.